Code/GraphMol/MolOps.cpp

```diff
@@ -383,6 +383,48 @@ void sanitizeMol(RWMol &mol, unsigned int &operationThatFailed,
   operationThatFailed = 0;
 }
 
+std::vector<std::unique_ptr<MolSanitizeException>> detectChemistryProblems(
+    const ROMol &imol, unsigned int sanitizeOps) {
+  RWMol mol(imol);
+  std::vector<std::unique_ptr<MolSanitizeException>> res;
+
+  // clear out any cached properties
+  mol.clearComputedProps();
+
+  int operation;
+  operation = SANITIZE_CLEANUP;
+  if (sanitizeOps & operation) {
+    // clean up things like nitro groups
+    cleanUp(mol);
+  }
+
+  // update computed properties on atoms and bonds:
+  operation = SANITIZE_PROPERTIES;
+  if (sanitizeOps & operation) {
+    for (auto &atom : mol.atoms()) {
+      try {
+        bool strict = true;
+        atom->updatePropertyCache(strict);
+      } catch (const MolSanitizeException &e) {
+        res.push_back(std::unique_ptr<MolSanitizeException>(e.copy()));
```

Comment:
This is esoteric, but using emplace_back isn't recommended when using vectors of unique_ptrs.  Apparently it can leak if the vector fails to allocate.  Of course, we're pretty hosed at that point :)

C++14 has make_unique which I believe is the preferred method these days, but we are using C++11 so the code as stands is ok I think.

Reply:
Oh, I didn't know about that quirk. Better not use it, then.

Reply:
Agreed that using `std::make_unique()` would be ideal, but we need to decide that we're really switching to C++14 to use that. I don't think making that switch would be a problem.

but it's not for this PR. :-)

Reply:
on the subject of @ricrogz's suggestion: it's definitely shorter and cleaner.
I think I'm willing to live with the leak in the case that expanding the vector fails. As @bp-kelley said: if that happens we're screwed anyway.
I'll switch to the `emplace_back` form and try to remember to use that in the future (I think I've said that before).